File: web/src/api/backend/system/sms.ts
/**
 * 短信管理 API（模板、变量、日志）
 */
import { adminRequest } from '@/utils/http'

// ===================== 短信模板 =====================

export interface SmsTemplateItem {
  id: number
  title: string
  code: string
  content: string
  providerTemplateId: string
  variables: any
  relatedVariableId: number
  status: number
  sort: number
  remark: string
  createTime: number
  updateTime: number
}

export function fetchSmsTemplateList(params: any) {
  return adminRequest.get<{ list: SmsTemplateItem[]; total: number }>({
    url: '/sms/template/list',
    params
  })
}

export function fetchSmsTemplateSave(params: any) {
  return adminRequest.post({ url: '/sms/template/save', params })
}

export function fetchSmsTemplateDelete(id: number) {
  return adminRequest.post({ url: '/sms/template/delete', params: { id } })
}

export function fetchSmsTemplateTest(params: { id: number; phone: string }) {
  return adminRequest.post<{ success: boolean; requestId: string; message: string }>({
    url: '/sms/template/test',
    params
  })
}

// ===================== 短信变量 =====================

export interface SmsVariableItem {
  id: number
  title: string
  name: string
  sourceType: number
  sqlQuery: string
  methodName: string
  sharedCount: number
  status: number
  createTime: number
  updateTime: number
}

export function fetchSmsVariableList(params: any) {
  return adminRequest.get<{ list: SmsVariableItem[]; total: number }>({
    url: '/sms/variable/list',
    params
  })
}

export function fetchSmsVariableSave(params: any) {
  return adminRequest.post({ url: '/sms/variable/save', params })
}

export function fetchSmsVariableDelete(id: number) {
  return adminRequest.post({ url: '/sms/variable/delete', params: { id } })
}

// ===================== 短信日志 =====================

export interface SmsLogItem {
  id: number
  phone: string
  templateCode: string
  driver: string
  content: string
  params: any
  status: number
  requestId: string
  errorMsg: string
  createTime: number
}

export function fetchSmsLogList(params: any) {
  return adminRequest.get<{ list: SmsLogItem[]; total: number }>({
    url: '/sms/log/list',
    params
  })
}

File: server/internal/model/input/adminin/sms.go
package adminin

// ==================== 短信模板 ====================

type SmsTemplateListInp struct {
	Page   int
	Size   int
	Status int
	Code   string
	Title  string
}

type SmsTemplateListModel struct {
	List  []SmsTemplateListItem `json:"list"`
	Total int                   `json:"total"`
	Page  int                   `json:"page"`
	Size  int                   `json:"size"`
}

type SmsTemplateListItem struct {
	Id                 uint64      `json:"id"`
	Title              string      `json:"title"`
	Code               string      `json:"code"`
	Content            string      `json:"content"`
	ProviderTemplateId string      `json:"providerTemplateId"`
	Variables          interface{} `json:"variables"`
	RelatedVariableId  uint64      `json:"relatedVariableId"`
	Status             int         `json:"status"`
	Sort               int         `json:"sort"`
	Remark             string      `json:"remark"`
	CreateTime         uint64      `json:"createTime"`
	UpdateTime         uint64      `json:"updateTime"`
}

type SmsTemplateSaveInp struct {
	Id                 uint64
	Title              string
	Code               string
	Content            string
	ProviderTemplateId string
	Variables          string
	RelatedVariableId  uint64
	Status             int
	Sort               int
	Remark             string
}

type SmsTemplateTestInp struct {
	Id    uint64
	Phone string
}

type SmsTemplateTestModel struct {
	Success   bool   `json:"success"`
	RequestId string `json:"requestId"`
	Message   string `json:"message"`
}

// ==================== 短信变量 ====================

type SmsVariableListInp struct {
	Page int
	Size int
	Name string
}

type SmsVariableListModel struct {
	List  []SmsVariableListItem `json:"list"`
	Total int                   `json:"total"`
	Page  int                   `json:"page"`
	Size  int                   `json:"size"`
}

type SmsVariableListItem struct {
	Id          uint64 `json:"id"`
	Title       string `json:"title"`
	Name        string `json:"name"`
	SourceType  int    `json:"sourceType"`
	SqlQuery    string `json:"sqlQuery"`
	MethodName  string `json:"methodName"`
	SharedCount int    `json:"sharedCount"`
	Status      int    `json:"status"`
	CreateTime  uint64 `json:"createTime"`
	UpdateTime  uint64 `json:"updateTime"`
}

type SmsVariableSaveInp struct {
	Id         uint64
	Title      string
	Name       string
	SourceType int
	SqlQuery   string
	MethodName string
	Status     int
}

// ==================== 短信日志 ====================

type SmsLogListInp struct {
	Page         int
	Size         int
	Phone        string
	TemplateCode string
	Status       int
	Driver       string
}

type SmsLogListModel struct {
	List  []SmsLogListItem `json:"list"`
	Total int              `json:"total"`
	Page  int              `json:"page"`
	Size  int              `json:"size"`
}

type SmsLogListItem struct {
	Id           uint64      `json:"id"`
	Phone        string      `json:"phone"`
	TemplateCode string      `json:"templateCode"`
	Driver       string      `json:"driver"`
	Content      string      `json:"content"`
	Params       interface{} `json:"params"`
	Status       int         `json:"status"`
	RequestId    string      `json:"requestId"`
	ErrorMsg     string      `json:"errorMsg"`
	CreateTime   uint64      `json:"createTime"`
}
